lib/ansible/modules/database/mysql/mysql_variables.py

```diff
@@ -186,10 +244,30 @@ def main():
         # Type values before using them
         value_wanted = typedvalue(value)
         value_actual = typedvalue(mysqlvar_val)
+        value_in_auto_cnf = None
+        if var_in_mysqld_auto_cnf is not None:
+            value_in_auto_cnf = typedvalue(var_in_mysqld_auto_cnf)
+
         if value_wanted == value_actual:
-            module.exit_json(msg="Variable already set to requested value", changed=False)
+            if mode == 'persist':
+                if value_in_auto_cnf is None:
+                    module.warn("The current global variable '%s' has value '%s' "
+                                "which corresponds to desired state but there is no value "
+                                "in mysqld-auto.cnf and it will be added only to there "
+                                "respectively." % (mysqlvar, value_actual))
+
+                elif value_in_auto_cnf is not None and value_wanted != value_in_auto_cnf:
+                    module.warn("The current global variable '%s' has value '%s' "
+                                "which corresponds to desired state but value in mysqld-auto.cnf "
+                                "is '%s' and will be changed only there "
+                                "respectively." % (mysqlvar, value_actual, value_in_auto_cnf))
+                else:
+                    module.exit_json(msg="Variable already set to requested value "
+                                         "and stored into mysqld-auto.cnf file", changed=False)
+            else:
+                module.exit_json(msg="Variable already set to requested value", changed=False)
```

Comment:
For `persist_only`, this behavior is wrong: you can only do this if `value_wanted == value_in_auto_cnf` in that case.